File: selenium-python/pages/form_page.py
from __future__ import annotations

from pages.base_page import BasePage


class FormPage(BasePage):
    email = "email"
    message = "message"
    submit = "submit-btn"
    status = "form-status"
    view = "view-form"

    def submit_form(self, email: str, message: str) -> None:
        self.type_into(self.email, email)
        self.type_into(self.message, message)
        self.click_on(self.submit)

    def expect_visible(self) -> None:
        assert self.is_test_id_visible(self.view)
        assert self.is_test_id_visible(self.email)
        assert self.is_test_id_visible(self.message)

    def get_status_text(self) -> str:
        return self.get_test_id_text(self.status)

File: selenium-java/src/test/java/frameworks/seleniumjava/pages/FormPage.java
package frameworks.seleniumjava.pages;

import org.openqa.selenium.WebDriver;
import io.qameta.allure.Step;
import static org.junit.jupiter.api.Assertions.assertTrue;

public final class FormPage extends BasePage {
  private static final String EMAIL = "email";
  private static final String MESSAGE = "message";
  private static final String SUBMIT = "submit-btn";
  private static final String STATUS = "form-status";
  private static final String VIEW = "view-form";

  public FormPage(WebDriver driver) {
    super(driver);
  }

  @Step
  public void submitForm(String emailValue, String messageValue) {
    typeInto(EMAIL, emailValue);
    typeInto(MESSAGE, messageValue);
    clickOn(SUBMIT);
  }

  @Step
  public void expectVisible() {
    assertTrue(isTestIdVisible(VIEW));
    assertTrue(isTestIdVisible(EMAIL));
    assertTrue(isTestIdVisible(MESSAGE));
  }

  @Step
  public String getStatusText() {
    return getTestIdText(STATUS);
  }
}
